### PR TITLE
RELATED: BB-1369 drill for derived from AM

### PR DESCRIPTION
**Pull request by yanlipnican**

### CODE REVIEW

**Comment by yanlipnican** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2839/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2220/ : SUCCESS in 2m 52s

File: src/factory/HeaderPredicateFactory.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import { AFM, Execution } from '@gooddata/typings';
 import {
     findMeasureByLocalIdentifier,
-    getMasterMeasureLocalIdentifier
+    getMasterMeasureLocalIdentifier, isDerivedMeasure
```
Comment (petrjanu) [inline src/factory/HeaderPredicateFactory.ts]: Wrap

File: src/factory/HeaderPredicateFactory.ts [hunk 2]
```diff
@@ -51,6 +52,25 @@ function arithmeticMeasureLocalIdentifierDeepMatch(
...
 
+function getArithmeticMeasureOperandIdentifiers(measure: AFM.IMeasure): string[] {
+    if (AFM.isArithmeticMeasureDefinition(measure.definition)) {
+        return measure.definition.arithmeticMeasure.measureIdentifiers;
+    }
+
+    return null;
+}
+
+function getDerivedMeasureMasterMeasureOperandIdentifiers(measure: AFM.IMeasure, afm: IAfm): string[] {
```
Comment (petrjanu) [inline src/factory/HeaderPredicateFactory.ts]: Use `AFM.IAfm` so you don't need to do `import IAfm = AFM.IAfm;` few lines above

Comment (yanlipnican) [reply src/factory/HeaderPredicateFactory.ts]: Much better

File: src/factory/tests/HeaderPredicateFactory.spec.ts [hunk 3]
```diff
@@ -446,6 +446,36 @@ describe('composedFromUri', () => {
             expect(predicate(measureHeaders.identifierBasedCompareArithmeticMeasure, context)).toBe(true);
         });
     });
+
+    describe('derived from AM', () => {
+        it('should match when derived PP from AM operand matches header uri', () => {
```
Comment (petrjanu) [inline src/factory/tests/HeaderPredicateFactory.spec.ts]: `should match when derived PP from AM matches header uri` ?

Comment (yanlipnican) [reply src/factory/tests/HeaderPredicateFactory.spec.ts]: 👍

**Comment by yanlipnican** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2844/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2224/ : SUCCESS in 2m 14s

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by yanlipnican** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2846/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2225/ : SUCCESS in 3m 23s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1168/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2847/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/690/ : UNSTABLE in 17m 30s

**Comment by yanlipnican** [comment]
extended check - storybook

**Comment by yanlipnican** [comment]
extended test - storybook

File: src/helpers/tests/afmHelper.spec.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { AFM } from '@gooddata/typings';
 import {
     getMasterMeasureObjQualifier,
     getMasterMeasureLocalIdentifier,
-    findMeasureByLocalIdentifier
+    findMeasureByLocalIdentifier, isDerivedMeasure
```
Comment (petrjanu) [inline src/helpers/tests/afmHelper.spec.ts]: Wrap

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1173/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2853/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/694/ : UNSTABLE in 18m 18s

**Comment by yanlipnican** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by petrjanu** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Pull request merged**